internal/services/apimanagement/api_management_named_value_resource.go

```diff
@@ -129,6 +130,10 @@ func resourceApiManagementNamedValueCreateUpdate(d *pluginsdk.ResourceData, meta
 		},
 	}
 
+	if parameters.Properties.KeyVault != nil && (parameters.Properties.Secret == nil || !*parameters.Properties.Secret) {
+		return fmt.Errorf("secret must be true when value_from_key_vault is set")
```

Comment:
We usually place property names between back ticks 
```suggestion
		return fmt.Errorf("`secret` must be true when `value_from_key_vault` is set")
```